src/main/logutils.ts

```diff
@@ -110,48 +165,153 @@ const tailFile = (path: string) => {
     tailHandler.on("line", function(data: string) {
         handleLogLine(data);
     });
-    
+
     tailHandler.on("error", function(error: unknown ) {
       console.log('ERROR: ', error);
     });
-}    
+}
 
 /**
+ * Splits a WoW combat line intelligently with respect to quotes,
+ * lists, tuples, and what have we.
+ *
+ */
+const splitLogLine = (line: string, maxSplits?: number): LogLine => {
+    const line_len = line.length
+    const list_items: string[][] = [];
+    const args_list: any[] = [];
+    let in_quote = false;
+    let open_lists = 0;
+    let value: any = '';
+
+    // Combat log line always has '<timestamp>  <line>' format,
+    // that is, two spaces between ts and line.
+    const tsIndexEnd = line.indexOf('  ');
+    const timestamp = line.substring(0, tsIndexEnd);
+
+    for (let ptr = tsIndexEnd + 2; ptr < line_len; ptr++) {
+        const c = line.charAt(ptr);
+        if (c === '\n') {
+            break;
+        }
+
+        if (in_quote) {
+            if (c === '"') {
```

Comment:
Does this logic need to be so heavily nested? Can we just pull this to the start and have:

```
if (c === '"') {
  in_quote = !in_quote;
  continue;
}
```

Reply:
Yes, unfortunately it does. Basically, the logic is:

- Are we inside a quoted string?
  - Yes: Is this the end of the quote?
    - Yes: Set flag false, continue loop
    - No: Do nothing, and let the current character be added to `value`
  - No: Check the current character aginst the special characters (which have no special meaning inside a quoted string)
    - Matches: Perform logic and contonue loop, or break out depending on the case
    - Doesn't match: Add to the current character to `value`

**If** I were to move it into the `case '"'` of the `switch` block, a comma, and brackets/parenthesis would suddenly have special meaning inside a quoted string.